pycyphal/transport/can/media/pythoncan/_pythoncan.py

```diff
@@ -286,6 +320,7 @@ async def send(self, frames: typing.Iterable[Envelope], monotonic_deadline: floa
 
     def close(self) -> None:
         self._closed = True
+        self._tx_queue.put(None)
```

Comment:
This should be moved into the try/catch block below and also we need to join the tx thread.